pkg/cmd/server/start/bootstrap_node.go

```diff
@@ -252,35 +260,42 @@ func (o NodeOptions) loadBootstrap(hostnames []string, nodeConfigDir string) err
 
 	// if we had a previous node-config.yaml, continue using it
 	if _, err2 := os.Stat(o.ConfigFile); err2 == nil {
-		if err == nil {
-			glog.V(2).Infof("Unable to load node configuration from the server: %v", err)
+		if err != nil {
+			glog.Warningf("Unable to load node configuration from the server: %v", err)
 		}
 		return nil
 	}
 
-	// if there is no node-config.yaml and no server config map, generate one
-	if kerrors.IsNotFound(err) {
-		glog.V(2).Infof("Generating a local configuration since no server config available")
-		nodeConfig, err := o.NodeArgs.BuildSerializeableNodeConfig()
-		if err != nil {
-			return err
-		}
-		if err := o.NodeArgs.MergeSerializeableNodeConfig(nodeConfig); err != nil {
-			return err
-		}
-		nodeConfig.ServingInfo.ServerCert.CertFile = "server.crt"
-		nodeConfig.ServingInfo.ServerCert.KeyFile = "server.key"
-		nodeConfig.ServingInfo.ClientCA = "node-client-ca.crt"
-		nodeConfig.MasterKubeConfig = "node.kubeconfig"
-		b, err := configapilatest.WriteYAML(nodeConfig)
-		if err != nil {
-			return err
-		}
-		if err := ioutil.WriteFile(o.ConfigFile, b, 0600); err != nil {
-			return err
-		}
-		return nil
+	switch {
```

Comment:
if you are suppose to bootstrap and you can't, I think you should fail, not use a local config instead.

Reply:
Fair.  Although should probably be based on whether you asked for a config or not.

Reply:
I'm torn on this.  An empty config map should result in a default.  A missing config map or typod map seems like an error.  But at the same time a 403 or other error (especially for the default config) seems borderline arbitrary.  A node having a config you didn't expect is bad, a node being down because of a correlated auth mixup also seems bad.